### ISSUE TITLE
Unable to create a system terminal

### ISSUE THREAD

**Issue by BuZZ-dEE** [ctongfei/progressbar#42]
I get the following error in the VSCode terminal and also in the Eclipse console using version 0.7.0.

> Jul 13, 2018 1:45:04 AM org.jline.utils.Log logr
WARNUNG: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
> 
> Read file lines  15% [=========>            ]  2044/10000 (0:00:00 / 0:00:00) 
> Read file lines 100% [======================] 10000/10000 (0:00:01 / 0:00:00) 

This is my code.:

```java
  List<String> lineList = new ArrayList<>();
  long numberOfLines = 0;
  try {
	  numberOfLines = Files.lines(file.toPath()).count();
  } catch (IOException e) {
    e.printStackTrace();
  }
  try (
    Stream<String> lines = Files.lines(file.toPath());
  ) {
    ProgressBar pb = new ProgressBar("Read file lines", numberOfLines);
    lines.forEach((line) -> {
      lineList.add(line);
      pb.step();
    });
    pb.close();
    lines.close();
  } catch (IOException e1) {
    e1.printStackTrace();
  }
```

**Comment by ctongfei** [comment]
Which version of ProgressBar are you using?

**Comment by BuZZ-dEE** [comment]
0.7.0

**Comment by ctongfei** [comment]
Is the progressbar actually running or is it not functioning properly?

**Comment by BuZZ-dEE** [comment]
It seems to me, that it does not work properly. If you see the above progress from 2 files. The first file progress stucks at 15%. But maybe my code snippet is also wrong?

**Comment by ctongfei** [comment]
Do you get the error in a normal console (e.g. macOS Terminal)? And your job finished in 1 second there might be problems with synchronizing the output stream.

```
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
```
This is a problem after JLine 3 where it actively detects the so-called "dumb" terminals. I'll find a way to suppress this information.

You could try ProgressBar `0.5.5` since it uses JLine 2, which should behave normally in these non-standard terminals (embedded in VSCode / Eclipse / IntelliJ etc.)

**Comment by ctongfei** [comment]
Solution: https://github.com/jline/jline3/issues/291 . Will be fixed in the next version.

**Comment by BuZZ-dEE** [comment]
Neither version 0.5.0:

Read file lines   9% │███████████████████████████████████████████████████████████████████████████████████████████████████████████████│  1508/10000 (0:00:00 / 0:00:00)

nor version 0.5.5:

    Read file lines   7% │███████████████████████████████████████████████████████████████████████████████████████████████████████████████│ 10000/10000 (0:00:00 / 0:00:00) 
Read file lines 100% │███████████████████████████████████████████████████████████████████████████████████████████████████████████████│ 10000/10000 (0:00:01 / 0:00:00) 

works for me.

```java
  List<String> lineList = new ArrayList<>();
  long numberOfLines = 0;
  try {
	  numberOfLines = Files.lines(insertStatementsFile.toPath()).count();
  } catch (IOException e) {
    e.printStackTrace();
  }
  try (
    Stream<String> lines = Files.lines(insertStatementsFile.toPath());
  ) {
    ProgressBar pb = new ProgressBar("Read file lines", (int) numberOfLines);
    pb.start();
    // lineList = lines.filter((line) -> {
    //   pb.step();
    //   return line != null;
    // }).collect(Collectors.toList());
    lines.forEach((line) -> {
      lineList.add(line);
      pb.step();
    });
    // pb.close();
    pb.stop();
    lines.close();
  } catch (IOException e1) {
    e1.printStackTrace();
  }
```

BTW: If you have `ide`-tags and use the tag `ide:vscode` on this issue, then you should also add the tag `ide:eclipse`.

**Comment by ctongfei** [comment]
I was not talking about the 2 bars -- I was talking about the following warning message:
```
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
```

This should go away with `0.5.5`.

As for the two bars, It is probably caused by some synchronization problems between output streams. This problem is harder to fix

**Comment by BuZZ-dEE** [comment]
I know, but there seems another issue. In the end I would like to get this running. ;) Should I open a new issue or where can I ask such questions?

**Comment by BuZZ-dEE** [comment]
> As for the two bars, It is probably caused by some synchronization problems between output streams. This problem is harder to fix

With version 0.5.0, I only have one bar, but the progress is wrong.

**Comment by ctongfei** [comment]
With extremely short jobs (less than 1 second), there are synchronization problems with the output stream. You are welcome to open a new issue to track this, but do not expect it to be fixed soon as I myself don't know how to fix it yet. The current progressbar runs completely fine with much larger jobs.

**Comment by ctongfei** [comment]
I think that this is fixed by #51.
